### PR TITLE
LPS-194953 Add Feature Flag

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://liferay.atlassian.net/browse/LPS-194953

### CODE REVIEW

**Comment by mateussgn** [comment]
ci:stop

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:stop

**Review by pedro-oliveira446 — APPROVED** [review]

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:object

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/ObjectValidationRuleEngineRegistryImpl.java [hunk 1]
```diff
@@ -109,7 +110,14 @@ protected void activate(BundleContext bundleContext) {
...
 
-				emitter.emit(key);
+				if (FeatureFlagManagerUtil.isEnabled("LPS-188898")) {
+					emitter.emit(key);
+				}
+				else if (!(objectValidationRuleEngine instanceof
+							FunctionObjectValidationRuleEngineImpl)) {
+
+					emitter.emit(key);
+				}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/ObjectValidationRuleEngineRegistryImpl.java:120]: Hi @mateussgn I believe if we reorder the conditional block like this it'll be more clear, what do you think?

````suggestion
				if (!(objectValidationRuleEngine instanceof FunctionObjectValidationRuleEngineImpl)) {
					emitter.emit(key);
				}
				else if (FeatureFlagManagerUtil.isEnabled("LPS-188898")) {
					emitter.emit(key);
				}
````

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/ObjectValidationRuleEngineRegistryImpl.java (hunk 1 above)
Comment (mateussgn) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/ObjectValidationRuleEngineRegistryImpl.java:120]: Yes, @MarinhoFeliphe.
I agree with you!
Thanks for see this!

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:sf

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

**Comment by mateussgn** [comment]
ci:forward:force

**Comment by mateussgn** [comment]
Directly sent to Brian: https://github.com/brianchandotcom/liferay-portal/pull/140478

**Comment by mateussgn** [comment]
ci:close

**Pull request closed without merging**